src/Desktop.Analyzers/Core/DoNotUseInsecureDTDProcessing.cs

```diff
@@ -0,0 +1,860 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Desktop.Analyzers.Common;
+using Analyzer.Utilities;
+
+namespace Desktop.Analyzers
+{
+    /// <summary>
+    /// Secure DTD processing and entity resolution in XML
+    /// </summary>
+    public abstract class DoNotUseInsecureDTDProcessingAnalyzer<TLanguageKindEnum> : DiagnosticAnalyzer where TLanguageKindEnum : struct
+    {
+        internal const string RuleId = "CA3075";
+        private const string HelpLink = "http://aka.ms/CA3075";
+        // Do not use insecure API:
+
+        // Do not use overloads that enable dtd processing, use one takes XmlReader instead.
+        // Matches FxCop warnings 3056,3057,3059, 3060, 3062, 3063, 3064, 3070, 3071, 3072
+        internal static DiagnosticDescriptor RuleDoNotUseUnsafeDtdProcessingOverloads = CreateDiagnosticDescriptor(SecurityDiagnosticHelpers.GetLocalizableResourceString(nameof(DesktopAnalyzersResources.DoNotUseDtdProcessingOverloadsMessage)),
```

Comment:
Thanks for unifying the description. But you still need to combine all the descriptor into one. This can be done by using "{0}" as message and generating the actual message when reporting the diagnostic. After making this change, please get rid of `CreateDiagnosticDescriptor` helper as well, as it is no longer needed :)